lib/Dialect/FIRRTL/Transforms/ExtractInstances.cpp

```diff
@@ -674,18 +682,26 @@ void ExtractInstancesPass::extractInstances() {
           // `circuitNamespace.newName(...)` and visit every module/instance in
           // the old NLA, remove the reference to the old NLA, and add
           // references for all the newly-created ones.
+          // prithayan: Updated to use the namespace to get a new name, which
+          // just works with the NLATable analysis, since no reference exists
+          // with the Instance Ops which needs to be updated.
```

Comment:
Cool, this is awesome! :100: That was a ticking time bomb :see_no_evil:

Reply:
Feel free to just delete the entire caveat comment I added since it no longer applies thanks to your fix.